app/services/user_seeder.rb

```diff
@@ -0,0 +1,122 @@
+class UserSeeder
+  def self.run(**opts)
+    new(**opts).run
+  end
+
+  def initialize(**options)
+    @csv_file = options[:csv_file]
+    file = File.read(csv_file)
+    @csv = CSV.parse(file, headers: true)
+    validate_csv_headers
+    @email_domain = options[:email_domain]
+    validate_email_domain
+    @deploy_env = options[:deploy_env] || LoginGov::Hostdata.env
+  rescue Errno::ENOENT
+    raise ArgumentError.new("#{csv_file} does not exist")
+  end
+
+  # Seed the users whose PII is specified in the CSV file. Raises an
+  # ArgumentError if an email address is already taken and uses a transaction to
+  # avoid partial completion.
+  #
+  # @return [Integer] the number of users created
+  def run
+    validate_environment
+
+    ActiveRecord::Base.transaction do
+      @csv.each_with_index do |row, i|
+        email = "user#{i}@#{email_domain}"
+        row['email_address'] = email
+        row['password'] = PASSWORD
+        ee = EncryptedAttribute.new_from_decrypted(email)
+
+        User.create!(email_fingerprint: ee.fingerprint) do |user|
+          codes = setup_user(user: user, ee: ee)
+          row['codes'] = codes.join('|')
+
+          personal_key = create_profile(user: user, row: row)
+          row['personal_key'] = personal_key
+        end
+      end
+    end
+
+    save_updated_csv
+
+    csv.length # the number of saved users
+  rescue ActiveRecord::RecordNotUnique
+    msg = "email domain #{email_domain} invalid - would overwrite existing users"
+    raise ArgumentError.new(msg)
+  end
+
+  private
+
+  PASSWORD = 'S00per Seekr3t'
```

Comment:
rubocop will probably complain that you you need to add `.freeze` or `# frozen_string_literals: true` to the top

Reply:
surprisingly it didn't... should I add it to be consistent?